translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+
+        if end_time < datetime.now(timezone.utc):
+            event_status = "completed"
+    return event_status
+
+
+
+
+
+
+
+# Parse COtrip alert to WZDx
+def parse_alert(alert, callback_function=None):
+    if not validate_alert(alert):
+        if callback_function:    #Note :a call back fucnction , which will trigger every time the invalid data is given
+            callback_function(alert)
+        return None
+    feature = {}
+    geometry = {}
+    geometry['type'] = "LineString"
+    geometry['coordinates'] = parse_polyline(alert['alert:Polyline'])
+
+    feature['type'] = "Feature"
+    properties = {}
+
+    #### I included a skeleton of the message, fill out all required fields and as many optional fields as you can. Below is a link to the spec page for a road event
+    #### https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/spec-content/objects/RoadEvent.md
+
+    # road_event_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['road_event_id'] = ''
+
+    # Event Type ['work-zone', 'detour']
+    properties['event_type'] = 'work-zone'
+
+    # data_source_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['data_source_id'] = ''
+
+    # start_date
+    properties['start_date'] = reformat_datetime(alert['alert:ReportedTime'])
+
+    # end_date
+    properties['end_date'] = reformat_datetime(alert.get('alert:ExpectedEndTime', ''))
+
+    # start_date_accuracy
+    properties['start_date_accuracy'] = "estimated"
+
+    # end_date_accuracy
+    properties['end_date_accuracy'] = "estimated"
+
+    # beginning_accuracy
+    properties['beginning_accuracy'] = "estimated"
+
+    # ending_accuracy
+    properties['ending_accuracy'] = "estimated"
+
+    # road_name
+    road_name = alert['alert:RoadName']
+    properties['road_name'] = road_name
+
+    # direction
+    Direction_map = {'North': 'northbound', 'South': 'southbound', 'West': 'westbound', 'East': 'eastbound'}
+ 
+    properties['direction'] = Direction_map.get(alert['alert:Direction'])
+
+    # vehicle impact
+
+    properties['vehicle_impact'] = 'unknown' #get_vehicle_impact(alert['alert:ClosureType'])
+
+    # Relationship
+    properties['relationship'] = {}
+
+    # lanes
+    properties['lanes'] = []
+
+    # road_name
+    properties['road_number'] = ""
+
+    # beginning_cross_street
+    properties['beginning_cross_street'] = ""
+
+    # beginning_cross_street
+    properties['ending_cross_street'] = ""
+
+    # event status
+    properties['event_status'] = get_event_status(alert['alert:ReportedTime'], alert.get('alert:ExpectedEndTime'))
+
+    # type_of_work
+    # maintenance, minor-road-defect-repair, roadside-work, overhead-work, below-road-work, barrier-work, surface-work, painting, roadway-relocation, roadway-creation
+    properties['types_of_work'] = []
+
+    # reduced speed limit
+    properties['reduced_speed_limit'] = 25
+
+    # workers present
+    properties['workers_present'] = False
```

Comment:
Can we leave this blank if we don't know whether workers are present or not? I do see descriptions saying flaggers present in the sample data.

Reply:
should be  fixed

Reply:
This seems to be the same in the `icone_translator.py` where we are hard coding to False. If we don't know the value, let's not set it.

Reply:
should be fixed now in icone_translator.py